### PR TITLE
Add output format flag to operator dump

### LINKED ISSUE #36568: Add output format flag to `operator dump`
(This is used to request new product features, please visit <https://discuss.istio.io> for questions on using Istio)

**Describe the feature request**
Add a `--output` flag to `istioctl operator dump`. For example, the `json` format output to let the content be more readable in some cases, like expressing lists.
**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Fix https://github.com/istio/istio/issues/36568

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by monsteredc** [comment]
@morvencao PTAL.

**Review by morvencao — COMMENTED** [review]
generally LGTM
Is there a filed issue for this?

**Comment by monsteredc** [comment]
> generally LGTM Is there a filed issue for this?

@morvencao Nope. Just realized that `operator dump` can have the same output format logic with `profile dump`

**Comment by morvencao** [comment]
Better to have an issue to track this.

**Comment by monsteredc** [comment]
@morvencao Just checked https://github.com/istio/istio/pull/24403, and I think this issue is relavent: https://github.com/istio/istio/issues/24271. `operator dump` will share the same kind of details to users as `profile dump`.

File: operator/cmd/mesh/operator-dump.go [hunk 1]
```diff
@@ -38,6 +40,8 @@ func addOperatorDumpFlags(cmd *cobra.Command, args *operatorDumpArgs) {
 	cmd.PersistentFlags().StringVarP(&args.common.manifestsPath, "charts", "", "", ChartsDeprecatedStr)
 	cmd.PersistentFlags().StringVarP(&args.common.manifestsPath, "manifests", "d", "", ManifestsFlagHelpStr)
 	cmd.PersistentFlags().StringVarP(&args.common.revision, "revision", "r", "", OperatorRevFlagHelpStr)
+	cmd.PersistentFlags().StringVarP(&args.common.outputFormat, "output", "o", yamlOutput,
+		"Output format: one of json|yaml|flags")
```
Comment (morvencao) [inline operator/cmd/mesh/operator-dump.go:44]: What's the use case for flags output format of operator dump?
For profile dump, we can see what we have set for each flag, but what's the purpose of flags output for operator dump?

Comment (monsteredc) [reply operator/cmd/mesh/operator-dump.go]: @morvencao Fixed. Thanks for the reminder! Also, just created a new issue https://github.com/istio/istio/issues/36568 to track the feature.

**Review by morvencao — APPROVED** [review]
thanks!

**Review by jacob-delgado — APPROVED** [review]

**Pull request merged**